### PR TITLE
feat: Update zksolc for system contracts

### PR DESCRIPTION
**Pull request by 0xVolosnikov**
# What ❔

## Checklist

- [ ] PR title corresponds to the body of PR (we generate changelog entries from PRs).
- [ ] Tests for the changes have been added / updated.
- [ ] Documentation comments have been added / updated.

### CODE REVIEW

File: system-contracts/hardhat.config.ts [hunk 1]
```diff
@@ -6,8 +6,8 @@ import "@nomiclabs/hardhat-ethers";
 import "hardhat-typechain";
 
 // This version of system contracts requires a pre release of the compiler
-const COMPILER_VERSION = "1.5.0";
-const PRE_RELEASE_VERSION = "prerelease-a167aa3-code4rena";
+const COMPILER_VERSION = "fix-output-directory"; // 1.5.5
+const PRE_RELEASE_VERSION = "72ea6d2";
 function getZksolcUrl(): string {
```
Comment (StanislavBreadless) [inline system-contracts/hardhat.config.ts:11]: this `getZksolcUrl` thing was needed before when the prerelease version had not yet been published, now we can remove it, could you please switch to just providing version number in hardhat config ts? similarly to how it is done in l2-contracts

Comment (0xVolosnikov) [reply system-contracts/hardhat.config.ts:11]: @StanislavBreadless we are using the pre-release version in this PR

Comment (0xVolosnikov) [reply system-contracts/hardhat.config.ts:11]: This prerelease fixes a bug with `--output-dir`

**Comment by 0xVolosnikov** [comment]
Several of `zksolc` updates contain breaking changes. We need to make updates to `zksync-era` as well

**Comment by 0xVolosnikov** [comment]
Waiting for 1.5.5

**Review by StanislavBreadless — APPROVED** [review]

**Pull request merged**